### ISSUE TITLE
Collectd error installing graphs

### ISSUE THREAD

**Issue by denhamgithub** [wiedehopf/graphs1090#56]
Instaled readsb for feeding ADSB Exchange. Also successfully installed were tar1090 and ADSB Exchange but I got collectd errors doing the install script about two weeks after installing readsb.

Toshiba Satellite L15 notebook running Linux Mint 20.3.

lsb_release -a
No LSB modules are available.
Distributor ID:	Linuxmint
Description:	Linux Mint 20.3
Release:	20.3
Codename:	una

From System Reports --
System:    Kernel: 5.15.0-33-generic x86_64 bits: 64 compiler: N/A Desktop: MATE 1.26.0 wm: marco 
dm: LightDM Distro: Linux Mint 20.3 Una base: Ubuntu 20.04 focal 
Machine:   Type: Laptop System: TOSHIBA product: Satellite L15-B v: PSKVGU-00G00E serial: <filter> 
           Chassis: type: 10 serial: <filter> 
           Mobo: TOSHIBA model: MA20 serial: <filter> UEFI [Legacy]: TOSHIBA v: 1.20 

Can't figure out how to attach the text file with the requested 40 lines so will paste.

cat eckert-adsb-graphs-error.txt 
Jul 03 17:05:49 satellite collectd[313149]: copying DB from disk to /run/collectd
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "syslog" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "rrdtool" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "table" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "interface" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "cpu" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "aggregation" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "match_regex" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "df" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: plugin "disk" successfully loaded.
Jul 03 17:05:49 satellite collectd[313165]: ERROR: dlopen("/usr/lib/collectd/python.so") failed: /usr/lib/collectd/python.so: undefined symbol: PyFloat_Type. The most common cause for this problem is missing dependencies. Use ldd(1) to check the dependencies of the plugin / shared object.
Jul 03 17:05:49 satellite collectd[313165]: dlopen("/usr/lib/collectd/python.so") failed: /usr/lib/collectd/python.so: undefined symbol: PyFloat_Type. The most common cause for this problem is missing dependencies. Use ldd(1) to check the dependencies of the plugin / shared object.
Jul 03 17:05:49 satellite collectd[313165]: plugin_load: Load plugin "python" failed with status 2.
Jul 03 17:05:49 satellite collectd[313165]: Found a configuration for the `python' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Jul 03 17:05:49 satellite collectd[313165]: Plugin python failed to handle option ModulePath, return code: -1
Jul 03 17:05:49 satellite collectd[313165]: Error: Parsing the config file failed!
Jul 03 17:05:49 satellite systemd[1]: collectd.service: Main process exited, code=exited, status=1/FAILURE
Jul 03 17:05:49 satellite collectd[313166]: writing DB from /run/collectd to disk
Jul 03 17:05:50 satellite collectd[313166]: writeback size on disk: 4.0K        /var/lib/collectd/rrd/localhost.tar.gz
Jul 03 17:05:50 satellite systemd[1]: collectd.service: Failed with result 'exit-code'.
Jul 03 17:05:50 satellite systemd[1]: Failed to start Statistics collection and monitoring daemon.
Jul 03 17:06:00 satellite systemd[1]: collectd.service: Scheduled restart job, restart counter is at 3076.
Jul 03 17:06:00 satellite systemd[1]: Stopped Statistics collection and monitoring daemon.
Jul 03 17:06:00 satellite systemd[1]: Starting Statistics collection and monitoring daemon...
Jul 03 17:06:00 satellite collectd[313235]: copying DB from disk to /run/collectd
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "syslog" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "rrdtool" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "table" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "interface" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "cpu" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "aggregation" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "match_regex" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "df" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: plugin "disk" successfully loaded.
Jul 03 17:06:00 satellite collectd[313251]: ERROR: dlopen("/usr/lib/collectd/python.so") failed: /usr/lib/collectd/python.so: undefined symbol: PyFloat_Type. The most common cause for this problem is missing dependencies. Use ldd(1) to check the dependencies of the plugin / shared object.
Jul 03 17:06:00 satellite collectd[313251]: dlopen("/usr/lib/collectd/python.so") failed: /usr/lib/collectd/python.so: undefined symbol: PyFloat_Type. The most common cause for this problem is missing dependencies. Use ldd(1) to check the dependencies of the plugin / shared object.
Jul 03 17:06:00 satellite collectd[313251]: plugin_load: Load plugin "python" failed with status 2.
Jul 03 17:06:00 satellite collectd[313251]: Found a configuration for the `python' plugin, but the plugin isn't loaded or didn't register a configuration callback.
Jul 03 17:06:00 satellite collectd[313251]: Plugin python failed to handle option ModulePath, return code: -1
Jul 03 17:06:00 satellite collectd[313251]: Error: Parsing the config file failed!
den@satellite ~/ADSB $ 

Dennis Eckert denhamlist at gmail.com
JUL 5 2022 1150AM Pacific

PS -- As a beginner, filing an error report this way appeared the proper way to submit.

**Comment by wiedehopf** [comment]
Have you tried the proposed fix?

**Comment by wiedehopf** [comment]
https://github.com/wiedehopf/graphs1090#ubuntu-20-fixes-symptom-collectd-errors-out-on-startup

you might need to change it to python3.9 ... but i think it's still 3.8

**Comment by denhamgithub** [comment]
Okay I can try that. I'll have to do a bit of research on what is involved in making the change from one python version to another.  Like, where do I look to make that change?

For example, there was a fairly long write up on collectd errors with Ubuntu 20.04, but it was considerably beyond my experience to understand what they were talking about.  Like collectd seems to be used for a number of applications, such as keeping a long record of battery charge/ recharge performance to track battery performance.  And when I do 'apropos collectd', there are no other applications named as doing the same sort of thing that collectd does.

1020am wed

On July 20, 2022, at 9:35 AM, wiedehopf ***@***.***> wrote:

https://github.com/wiedehopf/graphs1090#ubuntu-20-fixes-symptom-collectd-errors-out-on-startup

you might need to change it to python3.9 ... but i think it's still 3.8

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.￼Message ID: ***@***.***>

[ { ***@***.***": "<a href="http://schema.org">http://schema.org</a>", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "<a href="https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190503864">https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190503864</a>", "url": "<a href="https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190503864">https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190503864</a>", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "<a href="https://github.com">https://github.com</a>" } } ]

**Comment by wiedehopf** [comment]
First find out which python version is in use.

Just try the commands as they are ... if it doesn't work, you can runt he command to undo the change.

**Comment by denhamgithub** [comment]
OK, but I'd like to put this issue aside to concentrate on my key issue:  providing more value to ADSBx network.  I'm now using a quite limited discone antenna up only 10 feet on a Linux Mint quite-limited laptop in the front room.  Where I've installed most of your packages up to  adsbexchange Anywhere.

But I have in my bedroom the best arrangement -- a 32-foot tall antenna provided by RadarBox with their XRange 2 receiver (which is on a Pi in their box.  With bias-t pre-amp and SAW filter.)

Since I don't own it, I do not have any access to it through SSH.  I don't know the user name (although they have me on their network as PGRANRB500251).  I have tried guessing the user name and password but no luck.  Unless you know or have some hints?

It looks like your 'Feed data from generic receiver' would make this work?  I've tried nmap on that RBox Pi and I think they are most likely using Port 3005.  As you suggested in your instructions.  But perhaps you may know much more about the RBox parameters than I do.

Oh, not sure if this relevant.  But my ISP gives me two WAN circuits for about $50/month USD.   Both are fiber-to-the home, and 100 mBits/sec down and 100 mBits/sec. up.  So the RBox Pi on the good antenna in the bedroom is on a 192.168.1.0/24 LAN segment. And my front room laptop on Linux Mint 20.3 (based on the Ubuntu 20.04 with collectd problems) is on a 192.68.0.1 LAN segment.  It's easy for me to log on to either LAN segment through wireless.

If getting access to that RBox Pi works with your solutions, that would be great.  And then I could work on the other issues.  As a back stop, I have another laptop in the bedroom right next to the RBox Pi and Ubuntu 18.04 but haven't installed any of your solutions.  If we can't make accessing the RBox Pi work, then I would disconnect the good antenna with pre-amp and SAW filter and plug in to the Ubuntu 18.04. The RBox folks send me an automated e-mail only if -- after 6 hours -- they detect I'm not feeding them data.  I've asked them how much they would sell me that antenna for as long as I continue to feed them.  But they haven't answered yet.

So, to sum up, is there any way we can make that RBox Pi --under the current circumstances -- as the main feeder with your 'Feed data from generic receiver'? I'm disabled, move slow and on some days can't be on a computer at all due to long-standing back pain.  Then I'm stuck having to lay in bed all day.

Best regards, 

Dennis Eckert Ephrata WA 98823 (the dry part f WA State east of the Cascade Mtn Range)


On July 20, 2022, at 10:27 AM, wiedehopf ***@***.***> wrote:

First find out which python version is in use.

Just try the commands as they are ... if it doesn't work, you can runt he command to undo the change.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.￼Message ID: ***@***.***>

[ { ***@***.***": "<a href="http://schema.org">http://schema.org</a>", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "<a href="https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190555286">https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190555286</a>", "url": "<a href="https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190555286">https://github.com/wiedehopf/graphs1090/issues/56#issuecomment-1190555286</a>", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "<a href="https://github.com">https://github.com</a>" } } ]

**Comment by wiedehopf** [comment]
You have to ask them to enable beast output because it's not enabled by default.

If you think it's already there .... just edit the receiver_options in /etc/default/readsb as suggested here:
https://github.com/wiedehopf/adsb-scripts/wiki/Automatic-installation-for-readsb#remote-data-source-radarcape-or-similar-configure-net-only-and-a-receiver-with-another-ip-address-as-a-data-source

Instead of using the SDR, readsb will try and connect to whatever address / port you give it.
```
RECEIVER_OPTIONS="--net-only --net-connector 192.168.2.7,30005,beast_in"
```
While the configurationf for using an rtl-sdr would look like this (in case you want to go back):
```
RECEIVER_OPTIONS="--device 0 --device-type rtlsdr --gain -10 --ppm 0"
```

Restarting the service after changing config:
```
sudo systemctl restart readsb

```


Just check the log as described in the readme or check the interface if you're getting data.

Once you have that set up on your mx linux, the adsbexchange feed will just use that data.

**Comment by denhamgithub** [comment]
Graphs now works! Out of the blue, just started working before I took
my first cup of coffee.

HTOP reports system has been up for two days.  Had a Firefox tab set
to http://192.168.1.87/graphs1090/ And there they were.  Screen shot
attached.

I didn't attempt your Python suggestion.

Oh, and the results, like number of aircraft tracked, are laughable
--12?  I was trying out the whole 'suite' of readsb >>> tar1090 >>>
graphs1090 >>> collectd.

So, I made no changes. This did not work yesterday -- all day and the
last I checked the graphs1090 web page was about 4PM yesterday
.

And there were no system updates. System Monitor reports:

Release Linux Mint 20.3 Una 
64-bit Kernel Linux 5.15.0-41-generic x86_64
MATE 1.26.0

Minimal Toshiba Satellite 15 laptop
Intel® Celeron(R) CPU N2840 @ 2.16GHz × 2
Memory:  1.8 GiB

Any thoughts you have on the 'why' it works now would be helpful.

Dennis FRI JUL 22-2022 720AM Pacific
